integrations/api_repo_collaborator_test.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package integrations
+
+import (
+	"net/http"
+	"testing"
+
+	"code.gitea.io/gitea/models/perm"
+	repo_model "code.gitea.io/gitea/models/repo"
+	"code.gitea.io/gitea/models/unittest"
+	user_model "code.gitea.io/gitea/models/user"
+	api "code.gitea.io/gitea/modules/structs"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestAPIRepoCollaboratorPermission(t *testing.T) {
+	defer prepareTestEnv(t)()
+	user4 := unittest.AssertExistsAndLoadBean(t, &user_model.User{ID: 4}).(*user_model.User)
+	user20 := unittest.AssertExistsAndLoadBean(t, &user_model.User{ID: 20}).(*user_model.User)
+	repo1 := unittest.AssertExistsAndLoadBean(t, &repo_model.Repository{ID: 1}).(*repo_model.Repository)
+	repo1Owner := unittest.AssertExistsAndLoadBean(t, &user_model.User{ID: repo1.OwnerID}).(*user_model.User)
+
+	// Login as User2.
+	session := loginUser(t, repo1Owner.Name)
+	testCtx := NewAPITestContext(t, repo1Owner.Name, repo1.Name)
+
+	t.Run("RepoOwnerShouldBeAdmin", func(t *testing.T) {
+		req := NewRequestf(t, "GET", "/api/v1/repos/%s/%s/collaborators/%s/permission?token=%s", repo1Owner.Name, repo1.Name, repo1Owner.Name, testCtx.Token)
+		resp := session.MakeRequest(t, req, http.StatusOK)
+
+		var repoPermission api.RepoCollaboratorPermission
+		DecodeJSON(t, resp, &repoPermission)
+
+		assert.Equal(t, "owner", repoPermission.Permission)
```

Comment:
```suggestion
		assert.Equal(t, "owner", repoPermission.Permission)
		assert.Equal(t,  ... repoPermission.RoleName)
```
...
:point_down: and below

Reply:
well it now comes to the uncertainties that I had at the beginning of this PR.

Please see WhyHasUser5ReadAccess and WhyHasUser20ReadAccess tests

Reply:
It seems test-data is prefilled (or getting) with permissions for specific users, but I don't know which users

Reply:
test data is comming from https://github.com/go-gitea/gitea/tree/main/models/fixtures

in edgecases it could be the case, that the fixtures need a fix, as information that on real instances exist is missing ...

I'll have a look into it - just allow maintainers to edit the pull and i can push changes to this pull

Reply:
ok, thanks!

> I'll have a look into it - just allow maintainers to edit the pull and i can push changes to this pull

it is already allowed, at least that checkbox `Allow edits by maintainers` is checked.

Reply:
it could be an issue because it's a fork from an fork ... so `mgjm` is allowed but not me ... - github issues :/

Reply:
I'll file a pull against your repo if I found the issue

Reply:
oh I see, my bad, I can't change that now, without closing this PR.

> I'll file a pull against your repo if I found the issue

alright

btw. I don't have any experience in writing tests in go and I might use the test suite in a wrong way. `onGiteaRun` ?